### ISSUE TITLE
PyVRP philosophy and vision

### ISSUE THREAD

**Issue by wouterkool** [PyVRP/PyVRP#164]
Somewhat inspired by #157, I think it would be good to define a clear philosophy/vision for PyVRP and include this in the docs.

Personally, I really like the idea of PyVRP providing a set of general building blocks: LocalSearch (including various operators), crossovers and population classes, next to general tooling. While focused on genetic algorithms, I think we agree other components (like constructions or helpers for other meta-heuristic algorithms) could be added.

As such I see some parallels with e.g. PyTorch, which provides (building blocks for) models, optimizers, loss functions and data loaders but let's the user implement the training loop. There are higher level API's where you can just `train()` a model, but these are provided by separate libraries (skorch, fast.ai, ignite, etc.).

Now I think it's great if a ready to use GeneticAlgorithm is part of PyVRP, but I don't think it needs to support a lot of customization (like https://github.com/N-Wouda/PyVRP/issues/111 which are hard to get right for every use case), as for those users can just implement their own algorithm inspired by the examples.

With this, we should also reconsider the current terminology ('individual', 'educate'), which is strongly tied to genetic algorithms. I prefer to call the things what they are (solutions, local search) and not what role they plan inside the metaheuristic, as this may change. In my opinion, it is best to have a GeneticAlgorithm which takes an Educator (or EducatorClass) argument which can then be a LocalSearch.

I think the right philosophy also helps to isolate components related to the genetic algorithm (i.e. penalty manager) from the local search.

**Comment by leonlan** [comment]
I agree with your points. The main building blocks that we can offer as VRP library are _operators_, such as crossovers, local search, destroy and repair/constructive (LNS). With these I think cover pretty much everything needed for designing your own metaheuristic. 

We also have https://github.com/N-Wouda/ALNS which also implements many acceptance criteria and whose core algorithm can serve as a template for the creation of other single trajectory algorithms. 

We should indeed rename classes as well at some point to accommodate for general use: related #46.

**Comment by N-Wouda** [comment]
My take for the philosophy bit is that I like simple, and I don't think the current codebase is simple enough.

**Comment by N-Wouda** [comment]
I think time has largely settled this open issue, so I'm closing it as stale.
